### ISSUE TITLE
Numerical data type issue with generate_historical_report 

### ISSUE THREAD

**Issue by jadecodes8191** [jadecodes8191/buildingEnergyApi#45]
Task_zero ran without issues, but when I went to generate_historical_report, it threw an invalid index error on line 458, more specifically here:

`if type(all_temps_copy.loc[index_tuple]) == pd.Series:`

**Comment by jadecodes8191** [comment]
`index_tuple` is the room number followed by the highest temperature cast to an int. After looking through the debugger and realizing that there is no index of ('137', 73) since all the numbers were floats, I think the issue may be that the old format of data had already rounded all the temperatures to integers, while the new Metasys data is still made up of floats.

**Comment by jadecodes8191** [comment]
Update: it also seems to be throwing this error when I run it on the old, non-Metasys data...

**Comment by jadecodes8191** [comment]
Update: I tried to reset the database (just moved to a new table, didn't want to delete all the old data!) to include only the September Metasys data. But for some reason 1) the error is still happening and 2) there's still data from February in the table deep into the processing stage where the error happens. This could definitely be related.

**Comment by jadecodes8191** [comment]
Also just split up the data into pretask4 and posttask4 because the generate_historical_report file takes quite a bit of time to run its course and I'm currently only testing task 4.

**Comment by jadecodes8191** [comment]
After opening the new MetasysLog database using database viewing software and sorting by timestamp, it's clear that everything in that database is from September. This means the program is still feeding in old data from another database.

**Comment by jadecodes8191** [comment]
Just found the stray database -- DailyDatabase is being read from in generate_historical_report, but it isn't written to in generate_historical_report, convert_metasys_data, or task_zero. This means it's still using old data.

**Comment by jadecodes8191** [comment]
Oops, never mind, it is written to in generate_historical_report. Going to keep looking...

**Comment by jadecodes8191** [comment]
Actually, that was on the right track! Here was the issue:

Although I had been updating DailyDatabase, DailyTempDatabase, DailyCarbonDatabase, and FilteredT3Database (all from Task 3), because there were errors on Task 4, I never reached the end of Task 4 where I was able to clear the databases. For this reason, the old data from February was still in the databases. 

As a result of discovering the bug, I created and ran a separate file to clear the databases. Now I'm running it again to see if it works.
